state-chain/chains/src/sol/api.rs

```diff
@@ -251,11 +253,36 @@ impl<Environment: SolanaEnvironment> SolanaApi<Environment> {
 		message: Vec<u8>,
 		cf_parameters: Vec<u8>,
 	) -> Result<Self, SolanaTransactionBuildingError> {
+		// Verify the validity of the CCM message before building the call.
+		CcmValidityChecker::is_valid(
```

Comment:
I had a discussion with @albert-llimos regarding the ```is_valid()``` check here. This check is redundant here since we do the same check while doing request swap but I guess we want to have it as an extra safety layer. In this case I would add a comment mentioning that this check has already been done during the request swap but we do it again here as extra layer of safety. This would improve readability as someone might get confused why we do this check again.

Reply:
Yep Updated the comments here